upgrader.go

```diff
@@ -66,6 +66,11 @@ func (u *Upgrader) UpgradeInbound(ctx context.Context, t transport.Transport, ma
 }
 
 func (u *Upgrader) upgrade(ctx context.Context, t transport.Transport, maconn manet.Conn, p peer.ID, dir network.Direction) (transport.CapableConn, error) {
+	var stat network.Stat
+	if cs, ok := maconn.(network.ConnStat); ok {
+		stat = cs.Stat()
```

Comment:
It would be _nice_ if we could just call this on-demand. Not critical but the results from `Stat` may not always be constant.

Reply:
It would be nice indeed, but right now it wouldn't buy us much the way the swarm treats the stat object. That could change too of course if we want to expose dynamic stat information.